nvdaHelper/vbufBase/backend.cpp

```diff
@@ -36,11 +34,13 @@ void VBufBackend_t::initialize() {
 	int renderThreadID=GetWindowThreadProcessId((HWND)UlongToHandle(rootDocHandle),NULL);
 	LOG_DEBUG(L"render threadID "<<renderThreadID);
 	registerWindowsHook(WH_CALLWNDPROC,destroy_callWndProcHook);
-	registerWindowsHook(WH_CALLWNDPROC,renderThread_callWndProcHook);
-	LOG_DEBUG(L"Registered hook, sending message...");
-	SendMessage((HWND)UlongToHandle(rootDocHandle),wmRenderThreadInitialize,(WPARAM)this,0);
-	LOG_DEBUG(L"Message sent, unregistering hook");
-	unregisterWindowsHook(WH_CALLWNDPROC,renderThread_callWndProcHook);
+	auto func = [&] {
+		LOG_DEBUG(L"Calling renderThread_initialize on backend at "<<this);
+		this->renderThread_initialize();
+	};
+	LOG_DEBUG(L"Calling execInWindow");
+	execInWindow((HWND)UlongToHandle(rootDocHandle),func);

```

Comment:
I'm a little unsure if this is completely safe as it is a call from one instance of the CRT to another, using a c++ class or specific feature. I really have no evidence that it won't work, but just something we should keep an eye on in various operating systems.  For instance, we avoided ever passing c++ strings for this reason. At worse